File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.cs
public class Num9935
{
    public static void Main(string[] args)
    {
        string s = Console.ReadLine();
        string bomb = Console.ReadLine();
        
        int len = bomb.Length;
        Stack<char> stack = new Stack<char>();

        foreach (char ch in s)
        {
            stack.Push(ch);

            if (stack.Count >= len && stack.Peek() == bomb[len - 1])
            {
                bool match = true;
                char[] temp = new char[len];
                for (int i = len - 1; i>= 0; i--)
                {
                    char c = stack.Pop();
                    temp[i] = c;
                    if (c != bomb[i])
                    {
                        match = false;
                    }
                }

                if (!match)
                {
                    foreach (var c in temp)
                    {
                        stack.Push(c);
                    }
                }
            }
        }
        if (stack.Count == 0)
        {
            Console.WriteLine("FRULA");
        }
        else
        {
            char[] result = stack.ToArray();
            Array.Reverse(result);
            Console.WriteLine(string.Join("", result));
        }
    }
}

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.cc
#include <iostream>
#include <bits/stdc++.h>

using namespace std;
string s, t;
int main() {
    cin >> s >> t;
    
    stack<char> stk;
    
    for(char c : s)
    {
        stk.push(c);
        if(stk.size() >= t.size() && stk.top() == t[t.size() - 1])
        {
            string temp = "";
            for(int i = 0; i < t.size(); i++)
            {
                temp.push_back(stk.top());
                stk.pop();
            }
            reverse(temp.begin(), temp.end());
            
            if(temp != t)
            {
                for(char c1 : temp)
                {
                    stk.push(c1);
                }
            }
        }
    }
    if(stk.size() == 0) cout << "FRULA" << '\n';
    else
    {
        string temp = "";
        while(stk.size())
        {
            temp += stk.top();
            stk.pop();
        }
        reverse(temp.begin(), temp.end());
        cout << temp << '\n';
    }

    return 0;
}